### DISCUSSION TITLE
ATMega2560 with custom USB-TTL converter

### DISCUSSION THREAD

**Discussion by varneyc** [shanteacontrols/OpenDeck#59] [Q&A]
Hi again,

I have an Arduino Mega with WiFi board. The WiFi capabilities are not something I am particularly interested in utilising with my OpenDeck build (I simply had the board lying around), but it is unfortunately causing some headaches with the flashing process. Because the board uses a seperate CH340G USB-TTL converter as opposed to the regular ATMega16u2 processor, I understand I will need a custom YAML descriptor, but I am not finding any documentation on the usbLink keys and how they work. 

The ATMega2560 can be configured using dip switches to communicate directly with the CH340G USB-TTL converter. From my understanding, the ATMega2560 needs to be acting as a master and be directly communicating with the USB-TTL converter (since I cannot program the CH240G to be recognised as a slave device by the OpenDeck firmware). Could I theoretically just omit this block from the YAML descriptor: 
```  
usbLink:
    type: "slave"
    uartChannel: 0
```
and change `usb: false` to `usb: true`, or is there more to this process?

Thanks,

Connor.

**Comment by paradajz** [comment]
Arduino Mega clones without mega16u2 don't work with OpenDeck.
